docker/verify_version_matching.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+import re
+import sys
+from typing import Optional, List, Tuple
+
+
+if(sys.version_info[0] < 3 or sys.version_info[1] < 7):
+    print("This script requires python version 3.7 and above. Please make sure to run with the proper version. Aborting...")
+    sys.exit(1)
+
+_local_version_separators = re.compile(r"[\._-]")
+
+
+def parse_local_version(local: str) -> List[Optional[int]]:
+    """
+    Takes a string like abc.1.twelve and turns it into ("abc", 1, "twelve").
+    """
```

Comment:
I think we should better explain what that method do, to do what you stated in the docstring we can just use split('.)